### PR TITLE
Scalar RemovedShape events from ERROR -> NOTE

### PR DESCRIPTION
**Pull request by rchache**
*Description of changes:*
For scalar shapes (and non-enum strings), removing them is safe and therefore can be lowered from an ERROR to a NOTE

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-diff/src/main/java/software/amazon/smithy/diff/evaluators/RemovedShape.java [hunk 1]
```diff
@@ -19,19 +19,23 @@
...
  */
 public final class RemovedShape extends AbstractDiffEvaluator {
     @Override
     public List<ValidationEvent> evaluate(Differences differences) {
         return differences.removedShapes()
                 .filter(shape -> !shape.hasTrait(PrivateTrait.class))
                 .filter(shape -> !isMemberOfRemovedShape(shape, differences))
-                .map(shape -> error(shape, String.format("Removed %s `%s`", shape.getType(), shape.getId())))
+                .map(shape -> isScalarType(shape)
+                        ? note(shape, String.format("Removed %s `%s`", shape.getType(), shape.getId()))
```
Comment (kstich) [inline smithy-diff/src/main/java/software/amazon/smithy/diff/evaluators/RemovedShape.java:37]: Wonder if this should be a `warning` instead.

File: smithy-diff/src/main/java/software/amazon/smithy/diff/evaluators/RemovedShape.java [hunk 2]
```diff
@@ -40,4 +44,19 @@ private boolean isMemberOfRemovedShape(Shape shape, Differences differences) {
                 .filter(member -> !differences.getNewModel().getShapeIds().contains(member.getContainer()))
                 .isPresent();
     }
+
+    private boolean isScalarType(Shape shape) {
+        return shape.isIntegerShape()
```
Comment (kstich) [inline smithy-diff/src/main/java/software/amazon/smithy/diff/evaluators/RemovedShape.java:49]: Should exclude `intEnum` like is done for `enum` below.

File: smithy-diff/src/main/java/software/amazon/smithy/diff/evaluators/RemovedShape.java [hunk 3]
```diff
@@ -40,4 +44,19 @@ private boolean isMemberOfRemovedShape(Shape shape, Differences differences) {
...
+            || shape.isBigIntegerShape()
+            || shape.isBlobShape()
+            || shape.isBooleanShape()
+            || shape.isByteShape()
+            || shape.isDoubleShape()
+            || shape.isFloatShape()
+            || shape.isShortShape()
+            || shape.isTimestampShape()
+            || shape.isLongShape()
+            || (shape.isStringShape() && !shape.hasTrait(EnumTrait.class));
```
Comment (kstich) [inline smithy-diff/src/main/java/software/amazon/smithy/diff/evaluators/RemovedShape.java:60]: Should exclude the `enum` shape type as well.

**Review by kstich — CHANGES_REQUESTED** [review]
Some typing updates in `isScalarType` as well as resolving the `warning` vs `note` conversation.

File: smithy-diff/src/main/java/software/amazon/smithy/diff/evaluators/RemovedShape.java (hunk 1 above)
Comment (mtdowling) [reply smithy-diff/src/main/java/software/amazon/smithy/diff/evaluators/RemovedShape.java:37]: This really should be a warning. We need to bring this to people's attention but not fail the build, hence a warning.

**Review by mtdowling — APPROVED** [review]

**Review by kstich — APPROVED** [review]

**Pull request merged**